core/tests/protractor_utils/waitFor.js

```diff
@@ -156,6 +158,16 @@ var modalPopupToAppear = async function() {
     element(by.css('.modal-body')), 'Modal taking too long to appear.');
 };
 
+/**
+ * Check if a file has been downloaded
+ */
+var fileToBeDownloaded = async function(filename) {
+  var name = Constants.DOWNLOAD_PATH + '/' + filename;
+  browser.driver.wait(function() {
```

Comment:
Looking at other instances of this, we should probably `await` this statement.

Reply:
Done

Reply:
Actually, I won't be changing this. After the test the e2e test fails. I'm not exactly sure why (I took this download verification code from another PR), but I think that this works without the await.

Reply:
I don't see how adding the `await` could break this. `await` is just an identity function unless the value it receives is a Promise ([docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/await)). If `browser.driver.wait()` returns a promise, we should wait for that promise to be resolved, right?

The Selenium docs use `await` for explicit waits: https://www.selenium.dev/docs/site/en/webdriver/waits/#explicit-wait. Also, this GitHub issue in the protractor repo notes that there is a difference between `browser.driver.[something]` and `browser.[something]`, but it suggests that `browser.driver.[something]` is better, which is what you're doing already.

Without the `await`, I would expect the `browser.driver.wait` line to do essentially nothing. We would continue executing subsequent lines before the wait condition was satisfied. The test might still work in this situation as long as it takes long enough for the code to execute, but we wouldn't want to rely on that. One way to test this might be to try removing the `browser.driver.wait` and see if the test still works.

In what way does the test fail if you include the `await`?

Reply:
If I include the await, then I see the file download, but the test does not register that the file downloaded. I'm pretty unfamiliar with await and e2e testing, so I don't really know what the difference between having await and not having await is. 

The error looks like this:

```
1) Preferences should export account data
  - Failed: File was not downloaded!
  Wait timed out after 10001ms
```

Reply:
@U8NWXD would love your thoughts on this. I'm pretty unfamiliar so any guidance would be great!

Reply:
Here's my understanding of how `await` works here:

* `browser.wait` returns a `Promise` object immediately that won't be fulfilled until the function you pass returns True. The `await` forces the code execution to wait for that Promise to be fulfilled before executing subsequent code. Without the `await`, the subsequent code will execute even before the file is present.

To test this, you could try adding an expectation that `fs.existsSync(name)` is True after your `browser.wait`. I suspect that this expect will fail if you don't have the `await`.

Without the `await`, I think your test will always pass because the test will finish running before the `waitFor` times out (remember that without the `await` we don't wait for the `browser.wait`). Could you try deliberately breaking the test, for example by providing the wrong name for the downloaded file? My guess is that even this broken test will pass if you don't have the `await`.

Reply:
I see. It was an issue with the download path. Thanks so much for the direction in diagnosing this!